DeepSpeech.py

```diff
@@ -1169,6 +1213,10 @@ def _next_epoch(self):
                     # The current epoch should also have a validation part
                     self._epochs_running.append(Epoch(self._epoch, self._num_jobs_dev, set_name='dev', report=is_display_step))
 
+                    # State Machine flag for decision making regarding early stopping in next epoch function call
+                    self._dev = True
```

Comment:
Its a good point actually, i tried implementing it here but even after the early stopping was triggered there were few train epochs which were being reported. We are accumulating the loss when the case is dev anyway. So it won't report loss on incorrect dataset.